src/components/FormComponents/Sykmeldingsperiode/GradInput.tsx

```diff
@@ -0,0 +1,41 @@
+import { ErrorMessage, TextField } from '@navikt/ds-react';
+import { useController } from 'react-hook-form';
+
+import { SykmeldingFormValues } from '../../Sykmelding/SykmeldingForm';
+
+import styles from './GradInput.module.css';
+
+type FormName = `periode.${number}.grad`;
+
+export interface Props {
+    index: number;
+    name: FormName;
+}
+
+function GradInput({ index, name }: Props): JSX.Element {
+    const { field, fieldState } = useController<SykmeldingFormValues, FormName>({
+        name,
+        rules: {
+            validate: (value) => {
+                if (!value) {
+                    return 'Du må fylle inn grad.';
+                }
+                if (value < 20) {
+                    return 'Grad kan ikke være lavere enn 20 prosent.';
+                }
+                if (value > 100) {
+                    return 'Grad kan ikke være høyere enn 100 prosent.';
+                }
+            },
+        },
+    });
+
+    return (
+        <div className={styles.grad}>
+            <TextField id={`grad${index}`} label="Oppgi grad" type="number" onChange={field.onChange} />
```

Comment:
Vil ikke dette gjøre susen mtp. a11y? Så sparer man seg en ekstra prop!

```suggestion
            <TextField id={name} label="Oppgi grad" type="number" onChange={field.onChange} />
```